module/block_iterator/executor/executor.go

```diff
@@ -136,5 +135,37 @@ func commitAndCheckpoint(log zerolog.Logger, batch storage.Batch, iter module.Bl
 		return fmt.Errorf("failed to checkpoint iterator: %w", err)
 	}
 
+	startIndex, endIndex, nextIndex := iter.Progress()
+	progress := CalculateProgress(startIndex, endIndex, nextIndex)
+
+	log.Info().
+		Str("commit-dur", time.Since(start).String()).
+		Uint64("start-index", startIndex).
+		Uint64("end-index", endIndex).
+		Uint64("next-index", nextIndex).
+		Str("progress", fmt.Sprintf("%.2f%%", progress)).
+		Msg("batch committed")
+
+	metrics.ExecutionLastChunkDataPackPrunedHeight(nextIndex - 1)
+
 	return nil
 }
+
+// CalculateProgress calculates the progress of the iteration, it returns a percentage
+// of the progress. [0, 100]
+func CalculateProgress(start, end, current uint64) float64 {
+	if end <= start {
+		return 100.0 // If start == end, assume full completion
+	}
+	if current < start {
+		return 0.0 // If current is below start, assume 0%
+	}
+	if current > end {
+		return 100.0 // If current is above end, assume 100%
+	}
+	progress := float64(current-start) / float64(end-start) * 100.0
```

Comment:
Maybe adjust total block count to `end-start+1` because `end` is inclusive. 

```suggestion
	progress := float64(current-start) / float64(end-start+1) * 100.0
```